source/fundamentals/databases-collections.txt

```diff
@@ -96,15 +98,55 @@ against a series of filters during writes to a collection. You can
 specify these filters using the `ValidationOptions
 <{+api+}/apidocs/mongodb-driver-core/com/mongodb/client/model/ValidationOptions.html>`__
 class, which accepts a series of `Filters
-<{+api+}/apidocs/mongodb-driver-core/com/mongodb/client/model/Filters.html>`__
-that specifies the validation rules and expressions:
+<{+api+}/apidocs/mongodb-driver-core/com/mongodb/client/model/Filters.html>`__ instances
+that specify the validation rules and expressions:
 
 .. literalinclude:: /examples/generated/DatabaseCollectionsTest.snippet.validation.kt
    :language: kotlin
 
 For more information, see the server documentation for :manual:`document
 validation </core/document-validation>`.
 
+.. _db-coll-specify-return-type:
+
+Specify Return Type
+~~~~~~~~~~~~~~~~~~~
+
+The driver provides a way for you to specify a class for documents
+returned from a collection, even if it is different than the class you
+specified when retrieving the collection. You can specify a return class
+by using the `MongoCollection.withDocumentClass()
+<{+api-kotlin+}/apidocs/mongodb-driver-kotlin-coroutine/mongodb-driver-kotlin-coroutine/com.mongodb.kotlin.client.coroutine/-mongo-collection/with-document-class.html>`__
+method.
+
+Specifying a different return class could be useful in the following
+situations:
+
+- Your collection contains multiple data types.
+- You specify a projection that changes your data fields.
+- You cannot directly specify a return type on a method that changes the data,
+  such as ``findOneAndUpdate()`` or ``findOneAndReplace()``.
+
+The following example shows how you can retrieve a collection that
+contains represented by the ``Fruit`` data class but return the result
+of a ``findOneAndUpdate()`` operation as an instance of the ``NewFruit``
+class:
```

Comment:
suggestion: mention the renamed field? it honestly took me a second to realize the difference between `Fruit` and `NewFruit`

Reply:
added descriptions of the operation to both pages.